python/tests/integration/test_worker_recycling_integration.py

```
@@ -0,0 +1,135 @@
+
+
+def _receive_close_skipping_pings(websocket: SimpleWebSocketClient) -> tuple[int, str]:
+    """Receive frames until a close frame arrives (heartbeat pings may interleave)."""
+    for _ in range(10):
+        opcode, payload = websocket._receive_frame()
+        if opcode == 0x8:
+            if len(payload) >= 2:
+                return int.from_bytes(payload[:2], "big"), payload[2:].decode("utf-8")
+            return 1005, ""
+        assert opcode in (0x9, 0xA), f"Unexpected frame opcode {opcode} while waiting for close"
+    raise AssertionError("No close frame received")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)